### ISSUE TITLE
Metabase won't start on AWS, 0.25.1

### ISSUE THREAD

**Issue by jonahwh** [metabase/metabase#5664]
Just attempted to upgrade from 0.24.2 to 0.25.1, I get the following log output, then nothing. All requests fail with 504.

```
9863bda25322-stdouterr.log:  08-02 16:24:00 INFO util.encryption :: DB details encryption is ENABLED for this Metabase instance. ?
9863bda25322-stdouterr.log:  08-02 16:24:10 INFO metabase.core :: Starting Metabase in STANDALONE mode
9863bda25322-stdouterr.log:  08-02 16:24:10 INFO metabase.core :: Launching Embedded Jetty Webserver with config:
9863bda25322-stdouterr.log:   {:port 3000, :host "0.0.0.0"}
9863bda25322-stdouterr.log:  08-02 16:24:10 INFO metabase.core :: Starting Metabase version v0.25.1 (448c889 release-0.25.1) ...
9863bda25322-stdouterr.log:  08-02 16:24:10 INFO metabase.core :: System timezone is 'GMT' ...
9863bda25322-stdouterr.log:  08-02 16:24:10 DEBUG metabase.driver :: Registered driver :bigquery ?
9863bda25322-stdouterr.log:  08-02 16:24:10 DEBUG metabase.driver :: Registered driver :crate ?
9863bda25322-stdouterr.log:  08-02 16:24:10 DEBUG metabase.driver :: Registered driver :druid ?
9863bda25322-stdouterr.log:  08-02 16:24:10 WARN metabase.driver :: No -init-driver function found for 'metabase.driver.google'
9863bda25322-stdouterr.log:  08-02 16:24:10 DEBUG metabase.driver :: Registered driver :googleanalytics ?
9863bda25322-stdouterr.log:  08-02 16:24:10 DEBUG metabase.driver :: Registered driver :h2 ?
9863bda25322-stdouterr.log:  08-02 16:24:11 DEBUG metabase.driver :: Registered driver :mongo ?
9863bda25322-stdouterr.log:  08-02 16:24:11 DEBUG metabase.driver :: Registered driver :mysql ?
9863bda25322-stdouterr.log:  08-02 16:24:11 DEBUG metabase.driver :: Registered driver :postgres ?
9863bda25322-stdouterr.log:  08-02 16:24:11 DEBUG metabase.driver :: Registered driver :presto ?
9863bda25322-stdouterr.log:  08-02 16:24:11 DEBUG metabase.driver :: Registered driver :redshift ?
9863bda25322-stdouterr.log:  08-02 16:24:11 DEBUG metabase.driver :: Registered driver :sqlite ?
9863bda25322-stdouterr.log:  08-02 16:24:11 DEBUG metabase.driver :: Registered driver :sqlserver ?
9863bda25322-stdouterr.log:  08-02 16:24:11 INFO metabase.core :: Setting up and migrating Metabase DB. Please sit tight, this may take a minute...
9863bda25322-stdouterr.log:  08-02 16:24:11 INFO metabase.db :: Verifying postgres Database Connection ...
9863bda25322-stdouterr.log:  08-02 16:24:11 INFO metabase.db :: Verify Database Connection ...  ✅
9863bda25322-stdouterr.log:  08-02 16:24:11 INFO metabase.db :: Running Database Migrations...
9863bda25322-stdouterr.log:  08-02 16:24:11 INFO metabase.db :: Setting up Liquibase...
9863bda25322-stdouterr.log:  08-02 16:24:11 INFO metabase.db :: Liquibase is ready.
9863bda25322-stdouterr.log:  08-02 16:24:11 INFO metabase.db :: Checking if Database has unrun migrations...
9863bda25322-stdouterr.log:  08-02 16:24:14 INFO metabase.db :: Database has unrun migrations. Waiting for migration lock to be cleared...
9863bda25322-stdouterr.log:  08-02 16:24:14 INFO metabase.db :: Migration lock is cleared. Running migrations...
9863bda25322-stdouterr.log:  08-02 16:24:19 INFO metabase.db :: Database Migrations Current ...  ✅
9863bda25322-stdouterr.log:  com.mchange.v2.cfg.DelayedLogItem [ level -> FINE, text -> "The configuration file for resource identifier 'hocon:/reference,/application,/c3p0,/' could not be found. Skipping.", exception -> null]
9863bda25322-stdouterr.log:  08-02 16:24:19 INFO db.migrations :: Running all necessary data migrations, this may take a minute.
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO db.migrations :: Running data migration 'ensure-protocol-specified-in-site-url'...
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO db.migrations :: Finished running data migrations.
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.events :: Starting events listener: metabase.events.activity-feed ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.events :: Starting events listener: metabase.events.dependencies ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.events :: Starting events listener: metabase.events.driver-notifications ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.events :: Starting events listener: metabase.events.last-login ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.events :: Starting events listener: metabase.events.metabot-lifecycle ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.events :: Starting events listener: metabase.events.notifications ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.events :: Starting events listener: metabase.events.revision ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.events :: Starting events listener: metabase.events.sync-database ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.events :: Starting events listener: metabase.events.view-log ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.task :: Loading tasks namespace: metabase.task.follow-up-emails ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.task :: Loading tasks namespace: metabase.task.send-anonymous-stats ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.task :: Loading tasks namespace: metabase.task.send-pulses ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.task :: Loading tasks namespace: metabase.task.sync-databases ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.task :: Loading tasks namespace: metabase.task.upgrade-checks ?
9863bda25322-stdouterr.log:  08-02 16:24:20 INFO metabase.core :: Metabase Initialization COMPLETE
```

**Comment by salsakran** [comment]
The logs show that the metabase server is up and ready to accept connections.

A 504 indicates that a proxy somewhere in the request path isn't correct. 

What do you mean when you say "AWS" ... the beanstalk config or some other method?

Also note that if you're using ELBs (whether via the beanstalk config we share or manually), they will take an instance out of their server pool when the new version deploys, but should find them again after a few minutes.

**Comment by jonahwh** [comment]
It's running on beanstalk. I've never had this kind of delay when upgrading in the past, so I'm not sure where the problem might lie :)

**Comment by salsakran** [comment]
Can you check the nginx, and any other logs you can find? 

And do you have enhanced health monitoring enabled?

**Comment by camsaul** [comment]
@jonahwh ?

**Comment by jonahwh** [comment]
I was actually going to attempt to update to 0.25.2 last night, but there is no `metabase-aws-eb.zip` in the `downloads.metabase.com` S3 bucket for version 0.25.2, or 0.25.1 for that matter.

**Comment by salsakran** [comment]
@jonahwh that was a transient issue, and should be resolved now. Have you been able to figure out which proxy broke?

**Comment by hwartig** [comment]
I'll have the same issue. @jonahwh Did you find a solution to your problem?

When I'll try to update metabase (also running on aws eb) from v0.24.2 to any of the newer versions 0.25.1, 0.25.2 or 0.26.1 I cannot connect afterwards. Weirdly enough the `/api/health` endpoint still works.

<img width="809" alt="1__tmux" src="https://user-images.githubusercontent.com/84460/30925805-70c88dc6-a3b3-11e7-8629-d612d6a292d9.png">

Don't mind the socks proxy in the above screenshot I just wanted to make sure the aws load balancer isn't the issue so I'll tried connecting directly to the instance but I'll get the same behaviour when I go through the ELB.

In the nginx access log I see errors like the following
```
-------------------------------------
/var/log/nginx/access.log
-------------------------------------
10.0.0.207 - - [27/Sep/2017:16:30:55 +0000] "GET / HTTP/1.1" 444 0 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.101 Safari/537.36"
10.0.0.207 - - [27/Sep/2017:16:30:59 +0000] "GET /favicon.ico HTTP/1.1" 444 0 "http://ip-10-0-1-10.eu-west-1.compute.internal/api/health" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.101 Safari/537.36"
10.0.0.207 - - [27/Sep/2017:16:32:12 +0000] "GET / HTTP/1.1" 444 0 "-" "curl/7.54.0"
10.0.0.207 - - [27/Sep/2017:16:32:30 +0000] "GET / HTTP/1.1" 444 0 "-" "curl/7.54.0"
```

nothing in the other logs seems suspicious to me

**Comment by jonahwh** [comment]
I've finally gotten around to trying to upgrade again. I upgrade to v0.26.2, same issue. I can hit the health endpoint just fine, but any other request get a 444 from nginx, and a 504 to the client.

**Comment by ujjwal-gupta** [comment]
@jonahwh, @hwartig were you able to figure out what's going on?

**Comment by hwartig** [comment]
@ujjwal-gupta sadly not yet. I just gave up on upgrading for now. :-(

**Comment by si14** [comment]
Same here. @hwartig, @ujjwal-gupta, @jonahwh have you been able to sort it out? Quite annoying tbh :(

**Comment by ujjwal-gupta** [comment]
@si14 Sadly no. I'm stuck on 0.24.2. @salsakran Any upgrade path that works?

**Comment by si14** [comment]
Quick update for everyone interested: I gave up on sorting out the bug, but here is what I did:

* make an RDS snapshot from Metabase's own DB
* create second Beanstalk environment
* use the snapshot to create an RDS for the second environment
* manually copy env variables, https settings, etc. from the old environment
* "swap URLs" between environments

Basically I've just carefully recreated it all.

**Comment by ujjwal-gupta** [comment]
@si14 was this before or after upgrade to 0.27.2? i.e did you upgrade and then copy the non-functioning env or did you copy the environment and then upgrade to latest version?

**Comment by jonahwh** [comment]
I just want to point out @si14's workaround worked for me as well.

@ujjwal-gupta I created a new environment @ 0.27.2, then created a new database from the old snapshot.

**Comment by Glathrop** [comment]
@si14 i'm running into the same issue and wanted to take this approach but my initial beanstalk environment will not go into READY mode... you have to have two instances in ready to swap urls.... any thoughts?

**Comment by jonahwh** [comment]
I was able to rollback the old instance to `0.24.2` with no issue. Did the migration after that.

**Comment by Glathrop** [comment]
My original instance is screwed... wont restart properly and rebuilding will kill the db instances. I need have a cloned version that will build properly.... how can I just repoint the new instance DB to the old DB... I'm sniffing around env variables but

**Comment by hwartig** [comment]
I still had the same issue with a fresh environment. It seems to me that there is an issue due to the fact that I have `NGINX_FORCE_SSL` and `NGINX_SERVER_NAME` set at the same time. I took a look at the files included in the metabase-eb.zip.

```sh
######
# Metabase Report server Elastic Beanstalk configuration
# Modify the environmental variables below to customize your installation
# Comment out a variable to disable a feature
#####
container_commands:
    #customize_env: 
        #env:
            #NGINX_SERVER_NAME: metabase.example.com
            #NGINX_FORCE_SSL: 1
            #PAPERTRAIL_HOSTNAME: $HOSTNAME
            #PAPERTRAIL_HOST: foobar.papertrailapp.com
            #PAPERTRAIL_PORT: 12345
            #PAPERTRAIL_FILES: /var/log/nginx/access.log /var/log/nginx/error.log
        #command: true
        #ignoreErrors: false

    01_server-name:
        command: ".ebextensions/metabase_config/metabase-setup.sh server_name"
        test: test $NGINX_SERVER_NAME
        ignoreErrors: true

    02_server_https:
        command: ".ebextensions/metabase_config/metabase-setup.sh server_https"
        ignoreErrors: true

    03_log_x_real_ip:
        command: ".ebextensions/metabase_config/metabase-setup.sh log_x_real_ip"
        ignoreErrors: true
#...
```

This file instructs aws to first run `metabase-setup.sh server_name` and afterwards `metabase-setup.sh server_https`. Then I took a look at metabase-setup.sh and it seems that the `server_https` function accidentally undoes the work done by `server_name`.

So I went ahead and manually reran the metabase-setup.sh server_name command on the server. 

```sh
cd /var/app/current
sudo NGINX_SERVER_NAME=xyz.example.com .ebextensions/metabase_config/metabase-setup.sh server_name
sudo service nginx restart
```

This added the needed `server_name` instruction back to the nginx eb proxy configuration.

If you're starting with a fresh environment I think it should also work if you don't set a `NGINX_SERVER_NAME` environment variable in the first place. Then the setup scripts won't create the `/etc/nginx/sites-available/default-server`. This file is responsible for rejecting all traffic in case you have `NGINX_SERVER_NAME` set.

**Comment by vikeri** [comment]
@hwartig Your solution worked for me too!

**Comment by chadawagner** [comment]
https://github.com/metabase/metabase/pull/7204 moves server_https before server_name, which was my quick fix for this issue on our deployment

**Comment by jonahwh** [comment]
I had this same issue upgrading from v0.28.1 to v0.29.0.

I had to SET `NGINX_SERVER_NAME` in order for metabase to boot properly. Previously the env var was not set at all.

**Comment by Glathrop** [comment]
@si14 if there's any way you could expound on this it would be incredibly helpful. I still haven't figure out a great way to do this and the AWS instances crash for no reason about every four or five months.  

> use the snapshot to create an RDS for the second environment
> manually copy env variables, https settings, etc. from the old environment
> "swap URLs" between environments

From my understanding the steps are
```
#1 - Make sure you have a snapshot!
#2 aws rds delete --db-instance-identifier <value>
#3 aws rds restore-db-instance-from-db-snapshot --db-instance-identifier <value> --db-snapshot-identifier <value>
```

at this poing the db should be restored but I simply cannot get the Metabase EB environment to connect to the new db. Restarting the instance seems to crash everything, rebuilding environment makes a new RDS where the process would need to be repeated. Any assistance would be very helpful. I feel as if I’m simply missing a final step to force a db connection.

**Comment by flamber** [comment]
This was fixed around 0.30 - and there hasn't been any reports of something similar to this since.
Please comment with details if you're still experiencing this.
